File: frontend/src/services/categoryService.js
import api from './api';

// GET /api/category/all
export const getAllCategories = async () => {
    try {
        const response = await api.get('/category/all');
        return response.data;
    } catch (error) {
        console.error('Error fetching categories:', error);
        return [];
    }
};

// POST /api/category/add
export const addCategory = async (categoryData) => {
    try {
        const response = await api.post('/category/add', categoryData);
        return response.data;
    } catch (error) {
        console.error('Error adding category:', error);
        return { status: 'error', message: 'Failed to add category' };
    }
};

// POST /api/category/update
export const updateCategory = async (categoryData) => {
    try {
        const response = await api.post('/category/update', categoryData);
        return response.data;
    } catch (error) {
        console.error('Error updating category:', error);
        return { status: 'error', message: 'Failed to update category' };
    }
};

// POST /api/category/delete?id=1
export const deleteCategory = async (id) => {
    try {
        const response = await api.post(`/category/delete?id=${id}`);
        return response.data;
    } catch (error) {
        console.error('Error deleting category:', error);
        return { status: 'error', message: 'Failed to delete category' };
    }
};

File: backend/src/main/java/com/foodorder/service/CategoryService.java
package com.foodorder.service;

import com.foodorder.model.Category;
import com.foodorder.repository.CategoryRepository;

import java.sql.SQLException;
import java.util.List;

public class CategoryService {

    private CategoryRepository categoryRepository;

    public CategoryService() {
        this.categoryRepository = new CategoryRepository();
    }

    public String[] addCategory(Category category) {
        String[] result = new String[2];
        if (category.getName() == null || category.getName().trim().isEmpty()) {
            result[0] = "error"; result[1] = "Category name is required"; return result;
        }
        try {
            boolean saved = categoryRepository.save(category);
            result[0] = saved ? "success" : "error";
            result[1] = saved ? "Category added successfully" : "Failed to add category";
        } catch (SQLException e) {
            result[0] = "error"; result[1] = "Database error: " + e.getMessage();
        }
        return result;
    }

    public String[] updateCategory(Category category) {
        String[] result = new String[2];
        if (category.getId() <= 0) {
            result[0] = "error"; result[1] = "Invalid category ID"; return result;
        }
        if (category.getName() == null || category.getName().trim().isEmpty()) {
            result[0] = "error"; result[1] = "Category name is required"; return result;
        }
        try {
            boolean updated = categoryRepository.update(category);
            result[0] = updated ? "success" : "error";
            result[1] = updated ? "Category updated successfully" : "Failed to update category";
        } catch (SQLException e) {
            result[0] = "error"; result[1] = "Database error: " + e.getMessage();
        }
        return result;
    }

    public String[] deleteCategory(int id) {
        String[] result = new String[2];
        if (id <= 0) {
            result[0] = "error"; result[1] = "Invalid category ID"; return result;
        }
        try {
            boolean deleted = categoryRepository.delete(id);
            result[0] = deleted ? "success" : "error";
            result[1] = deleted ? "Category deleted successfully" : "Category not found";
        } catch (SQLException e) {
            result[0] = "error"; result[1] = "Database error: " + e.getMessage();
        }
        return result;
    }

    public List<Category> getAllCategories() {
        try {
            return categoryRepository.findAll();
        } catch (SQLException e) {
            e.printStackTrace(); return null;
        }
    }
}
